src/server/game/AI/CreatureAI.h

```diff
@@ -218,4 +219,51 @@ enum Permitions
     PERMIT_BASE_SPECIAL            = 800
 };
 
+class EvadeHomeCall : public BasicEvent
+{
+public:
+    explicit EvadeHomeCall(Creature* owner) : _owner(owner) { }
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/)
+    {
+        if (_owner->IsAIEnabled && _owner->GetAI() && reinterpret_cast<const SmartAI*>(_owner->GetAI()) == nullptr)
```

Comment:
I guess this should be a dynamic_cast, not a reinterpret one. That's evil, very evil.